### PR TITLE
Additional Actions

### PR DESCRIPTION
**Pull request by littleyoda**
I added quite a lot of features
-  CV-Read von POM via RailCom
- RailCom-Status
- CV-Write via PT and POM
- Testing classes
- time of keepalivetimer is adjustable

### CODE REVIEW

**Comment by grizeldi** [comment]
That looks awesome to me! I'm a fan of POM programming so this will come in handy. I'll test it in the following days.
Btw do you have any idea how could we implement testing using JUnit? Since it involves networking I'm not really sure we can...

**Comment by grizeldi** [comment]
Also I'll have to rely on you regarding everything involving LocoNet, because I have the white z21 which doesn't have the LocoNet socket.

File: src/main/java/z21Drive/Z21.java [hunk 1]
```diff
@@ -18,6 +9,28 @@
 import java.util.List;
 import java.util.logging.Logger;
 
+import javax.swing.Timer;
+
+import z21Drive.actions.Z21Action;
+import z21Drive.actions.Z21ActionGetSerialNumber;
+import z21Drive.actions.Z21ActionLanLogoff;
+import z21Drive.broadcasts.BroadcastTypes;
```
Comment (grizeldi) [inline src/main/java/z21Drive/Z21.java:17]: Can we please keep wildacrds? I know that it's not perfect, but the code is waaaaaay more readable that way.

File: src/main/java/z21Drive/testing/WriteCV.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * 
+ * BE CAREFUL!
+ * 
+ * This Class writes a new Loco ID to CV 1
+ * 
+ * BE CAREFUL!
+ * 
+ * 
+ */
+public class WriteCV implements Runnable{
```
Comment (grizeldi) [inline src/main/java/z21Drive/testing/WriteCV.java:24]: Maybe rename this to WriteLocoAdress?
Just an idea I got.

File: src/main/java/z21Drive/Z21.java (hunk 1 above)
Comment (littleyoda) [inline src/main/java/z21Drive/Z21.java:17]: I thing Eclipse has automaticlly removed the wildcards. 
I will disable this.

**Comment by littleyoda** [comment]
I have also "only" the white z21.

I pushed my Tool to github:
https://github.com/littleyoda/ly-Z21-Tool

It reads the loco decoder and show you the decoded content.
Unfortunately only in German and currently only for ESU decoder.

File: src/main/java/z21Drive/Z21.java (hunk 1 above)
Comment (grizeldi) [inline src/main/java/z21Drive/Z21.java:17]: Hmm... It's interesting how Intellij Idea prefers wildcards  and Eclipse prefers lots of imports.

**Comment by grizeldi** [comment]
If you want, I can add some Lenz capabilities to it ;)

**Comment by grizeldi** [comment]
Oh, wait. I'm stupid. I just messed up RailCom and LocoNet (facepalm).
Yes, of course I can test RailCom stuff ;)

**Comment by grizeldi** [comment]
Also can you port strings for your app into a resource bundle if you haven't already? I do feel like putting my limited German knowledge to a test and translating your app.

**Comment by littleyoda** [comment]
Can you send me an e-mail?
sb-www0703-xuzwshf74k8qzgyg@bursch.com

File: src/main/java/z21Drive/Z21.java [hunk 3]
```diff
@@ -30,7 +43,7 @@
     private List<Z21ResponseListener> responseListeners = new ArrayList<Z21ResponseListener>();
     private List<Z21BroadcastListener> broadcastListeners = new ArrayList<Z21BroadcastListener>();
     private DatagramSocket socket;
-    private static final int keepAliveTimeout = 30000;
+	private Timer keepAliveTimer;
```
Comment (grizeldi) [inline src/main/java/z21Drive/Z21.java:46]: This can be final, right?

Comment (littleyoda) [inline src/main/java/z21Drive/Z21.java:46]: yes. Can be final.

File: src/main/java/z21Drive/responses/Z21ResponseRailcomDatachanged.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Handels a RailCom-Response
+ * @author sven
+ *
+ */
+public class Z21ResponseRailcomDatachanged extends Z21Response {
```
Comment (grizeldi) [inline src/main/java/z21Drive/responses/Z21ResponseRailcomDatachanged.java:11]: Indentation of this class looks like it's gone wild. Please fix it.

Comment (littleyoda) [inline src/main/java/z21Drive/responses/Z21ResponseRailcomDatachanged.java:11]: done

**Comment by grizeldi** [comment]
Also your tests have some problems now, because some days ago I changed listener/event delivery thread to be daemon, so now your tests just exit...

**Comment by grizeldi** [comment]
Done! I hope it helps you in your project.

**Comment by littleyoda** [comment]
It will.

thanks.

**Pull request merged**